#!/usr/bin/env python
PACKAGE = "ueye_rad_converter"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("num_bins",    int_t,    0, "Number of histogram bins for autorange", 1000, 0, 10000)
gen.add("auto_range",   bool_t,    0, "Enable/Disable Autorange", True)
gen.add("min_value", int_t, 0, "Minimum intensity to map",    0, 0, 65535)
gen.add("max_value",    int_t,    0, "Maximum intensity to map", 65535, 0, 65535)
gen.add("hist_equal",    bool_t,    0, "Maximum intensity to map", True)


colormap_enum = gen.enum([ gen.const("AUTUMN",      int_t, 0, "Autumn colormap"),
                       gen.const("BONE",     int_t, 1, "Bone colormap"),
                       gen.const("JET",      int_t, 2, "Jet colormap"),
                       gen.const("WINTER", int_t, 3, "Winter colormap"),
                       gen.const("RAINBOW",     int_t, 4, "Rainbow colormap"),
                       gen.const("OCEAN",      int_t, 5, "Ocean colormap"),
                       gen.const("SUMMER", int_t, 6, "Summer colormap"),
                       gen.const("SPRING",     int_t, 7, "Spring colormap"),
                       gen.const("COOL",      int_t, 8, "Cool colormap"),
                       gen.const("HSV", int_t, 9, "HSV colormap"),
                       gen.const("PINK",     int_t, 10, "Pink colormap"),
                       gen.const("HOT",      int_t, 11, "Hot colormap"),
                       gen.const("IRON",      int_t, 12, "Iron colormap")],
                     "An enum of colormaps")

gen.add("color_map",   int_t,   0, "Colormap to use",  12, 0, 12, edit_method=colormap_enum)

exit(gen.generate(PACKAGE, "ueye_rad_converter", "UeyeRad"))
